### PR TITLE
Add timestamp to ECL print statements,

### PR DESCRIPTION
**Pull request by kamilritz**
which is helpful for EKF replay debugging.

@priseborough Is `_imu_sample_delayed.time_us` the right time to stamp the messages?

### CODE REVIEW

**Comment by dagar** [comment]
Can you do this without double precision math? These print statements are already relatively expensive on real hardware (stm32f4).

**Comment by kamilritz** [comment]
@dagar Double multiplication is now removed. The time is now printed in microseconds.

**Comment by kamilritz** [comment]
@jkflying Thanks, this is now resolved

**Comment by dagar** [comment]
> @dagar Double multiplication is now removed. The time is now printed in microseconds.

You're still casting to double though?

**Comment by mhkabir** [comment]
@kamilritz Just print the microseconds timestamp using `%llu` without casting.

**Comment by kamilritz** [comment]
@mhkabir Are you sure that this is working on all platforms. To me it seemed that some platforms interpret uint64_t as long unsigned int (e.g in EKF replay - make px4_sitl none) and some as long long unsigned int.

**Comment by mhkabir** [comment]
You're right - PRIu64 should be more portable. https://en.cppreference.com/w/c/types/integer

**Comment by kamilritz** [comment]
@dagar @mhkabir It is now using uint64 directly and is adding the timestamps in posix builds only.

File: EKF/gps_yaw_fusion.cpp [hunk 1]
```diff
@@ -174,7 +174,7 @@ void Ekf::fuseGpsAntYaw()
 
 		// we reinitialise the covariance matrix and abort this fusion step
 		initialiseCovariance();
-		ECL_ERR("EKF GPS yaw fusion numerical error - covariance reset");
+		ECL_ERR("%.0lf: EKF GPS yaw fusion numerical error - covariance reset", (double)_imu_sample_delayed.time_us);
```
Comment (jkflying) [inline EKF/gps_yaw_fusion.cpp]: Missed this one

**Comment by mhkabir** [comment]
I'm not fully sold on the idea using the delayed sample's timestamp. I feel like a log message should be using the actually current time (like hrt_absolute_time(), as opposed to one which can be delayed based on the fusion horizon.

**Comment by jkflying** [comment]
The issue is that during replay using the hrt_absolute_time() gives system time, which is useless.

**Comment by priseborough** [comment]
@kamilritz Can you please push a patch for the conflict.

File: EKF/control.cpp [hunk 2]
```diff
@@ -181,21 +180,22 @@ void Ekf::controlExternalVisionFusion()
...
-					ECL_INFO("EKF commencing external vision velocity fusion");
+					ECL_INFO_TIMESTAMPED("EKF commencing external vision velocity fusion");
 				}
 
 				if ((_params.fusion_mode & MASK_ROTATE_EV) && !(_params.fusion_mode & MASK_USE_EVYAW)
 					&& !_ev_rot_mat_initialised)  {
 					// Reset transformation between EV and EKF navigation frames when starting fusion
 					resetExtVisRotMat();
 					_ev_rot_mat_initialised = true;
+          ECL_INFO_TIMESTAMPED("EKF external vision aligned");
```
Comment (mhkabir) [inline EKF/control.cpp]: Indent

File: EKF/control.cpp [hunk 3]
```diff
@@ -1682,7 +1682,7 @@ void Ekf::controlVelPosFusion()
 				_fuse_hpos_as_odom = false;
 
 				if (_time_last_fake_gps != 0) {
-					ECL_WARN("EKF stopping navigation");
+				ECL_WARN_TIMESTAMPED("EKF stopping navigation");
```
Comment (mhkabir) [inline EKF/control.cpp]: Indent

Comment (kamilritz) [reply EKF/control.cpp]: thanks

**Comment by kamilritz** [comment]
@mhkabir @priseborough LGTM

**Comment by kamilritz** [comment]
@priseborough @julianoes @bkueng it is squashed now, could someone please merge it. Thanks!

**Pull request merged**